src/components/Dashboard/BorrowLimit.tsx

```diff
@@ -29,23 +27,7 @@ const CardWrapper = styled.div`
 const format = commaNumber.bindWith(',', '.');
 
 function BorrowLimit() {
-  const [available, setAvailable] = useState('0');
-  const [borrowPercent, setBorrowPercent] = useState(0);
-  const { account } = useWeb3Account();
-  const { userTotalBorrowBalance, userTotalBorrowLimit } = useMarketsUser();
-
-  useEffect(() => {
-    if (account) {
-      const total = BigNumber.maximum(userTotalBorrowLimit, 0);
-      setAvailable(total.dp(2, 1).toString(10));
-      setBorrowPercent(
-        total.isZero() || total.isNaN()
-          ? 0
-          : userTotalBorrowBalance.div(total).times(100).dp(0, 1).toNumber(),
-      );
-    }
-  }, [userTotalBorrowBalance, userTotalBorrowLimit]);
-
+  const { available, borrowPercent } = useBorrowLimit();
```

Comment:
just moved current calculations into hooks in order to reuse it before the logics refactoring